### PR TITLE
qt6, test: Handle deprecated code

### PR DESCRIPTION
**Pull request by hebasto**
Split from https://github.com/bitcoin/bitcoin/pull/30997.

This PR ensures compatibility across all supported Qt versions.

---

This PR can be tested on macOS using the first commit from https://github.com/bitcoin/bitcoin/pull/30997 and Homebrew's `qt` package.

### CODE REVIEW

**Comment by hebasto** [comment]
Friendly ping @Sjors @furszy @ryanofsky @jarolrod ;)

File: src/qt/test/rpcnestedtests.cpp [hunk 1]
```diff
@@ -127,13 +127,26 @@ void RPCNestedTests::rpcNestedTests()
     RPCConsole::RPCExecuteCommandLine(m_node, result, "rpcNestedTest(   abc   ,   cba )");
     QVERIFY(result == "[\"abc\",\"cba\"]");
 
+#if (QT_VERSION >= QT_VERSION_CHECK(6, 3, 0))
+    QVERIFY_THROWS_EXCEPTION(std::runtime_error, RPCConsole::RPCExecuteCommandLine(m_node, result, "getblockchaininfo() .\n")); //invalid syntax
```
Comment (promag) [inline src/qt/test/rpcnestedtests.cpp:131]: 2b5fd8fd24026ecf2fac7af6657e2363bfe3706c

An alternative to reduce duplicate code and the diff is to define `QVERIFY_EXCEPTION_THROWN`, something like?

```cpp
#if (QT_VERSION >= QT_VERSION_CHECK(6, 3, 0))
#define QVERIFY_EXCEPTION_THROWN(a, b) QVERIFY_THROWS_EXCEPTION(b, a)
#endif
```

**Review by promag — COMMENTED** [review]
Concept ACK

File: src/qt/test/rpcnestedtests.cpp (hunk 1 above)
Comment (hebasto) [reply src/qt/test/rpcnestedtests.cpp]: Thanks! Reworked per your feedback.

**Review by promag — COMMENTED** [review]
Code review ACK 5625840c11db2065a1c8d8de3babb6466eda04d4.

File: src/qt/test/rpcnestedtests.cpp [hunk 2]
```diff
@@ -127,6 +127,11 @@ void RPCNestedTests::rpcNestedTests()
     RPCConsole::RPCExecuteCommandLine(m_node, result, "rpcNestedTest(   abc   ,   cba )");
     QVERIFY(result == "[\"abc\",\"cba\"]");
 
+// Handle deprecated macro, can be removed once minimum Qt is at least 6.3.0.
+#if (QT_VERSION >= QT_VERSION_CHECK(6, 3, 0))
+#undef QVERIFY_EXCEPTION_THROWN
+#define QVERIFY_EXCEPTION_THROWN(expression, exceptiontype) QVERIFY_THROWS_EXCEPTION(exceptiontype, expression)
```
Comment (Sjors) [inline src/qt/test/rpcnestedtests.cpp:133]: 5625840c11db2065a1c8d8de3babb6466eda04d4: note to other reviewers: https://doc.qt.io/qt-6/qtest.html#QVERIFY_THROWS_EXCEPTION

**Review by Sjors — APPROVED** [review]
tACK 5625840c11db2065a1c8d8de3babb6466eda04d4

I only tested on Intel macOS 15.0 with Homebrew qt@5. Checked that the warnings still appear with `build/src/qt/test/test_bitcoin-qt`.

**Pull request merged**